support-models/src/main/scala/com/gu/support/zuora/api/PaymentGateway.scala

```diff
@@ -14,12 +14,6 @@ sealed trait PaymentGateway {
 }
 
 object PaymentGateway {
-  def forPaymentMethod(paymentMethod: PaymentMethod, currency: Currency): PaymentGateway = paymentMethod match {
-    case _: PayPalReferenceTransaction => PayPalGateway
-    case _: CreditCardReferenceTransaction if currency == AUD => StripeGatewayAUD
-    case _: CreditCardReferenceTransaction => StripeGatewayDefault
-    case _: DirectDebitPaymentMethod | _: ClonedDirectDebitPaymentMethod => DirectDebitGateway
-  }
```

Comment:
this is redundant as we store it on the object using inheritance instead of pattern matching